source/Tests/MessageContractAnalyzerFixture.cs

```diff
@@ -276,78 +276,100 @@ public class SimpleResponse : IResponse { }
+        public string InitializedStringProperty { get; set; } =  ""foobar""; // OK
 
         [Optional]
-        public bool BoolProperty { get; set; } // should NOT fire on this because we have a special case for bool
-    }
-    /// <summary>response</summary>
-    public class SimpleResponse : IResponse { }
-}");
+        public string InitializedStringPropertyEmpty { get; set; } = string.Empty; // OK
 
-            await Verify.VerifyAnalyzerAsync(source,
-                new DiagnosticResult(Descriptors.OptionalPropertiesOnMessageTypesMustBeNullable).WithLocation(0).WithArguments("StringProperty", "string"),
-                new DiagnosticResult(Descriptors.OptionalPropertiesOnMessageTypesMustBeNullable).WithLocation(1).WithArguments("IntProperty", "int"));
-        }
+        [Optional]
+        public string InitializedStringPropertyExplicitNull { get; set; } = null!; // OK; If you really want to break things, then you can.
 
-        [Test]
-        public async Task MessageTypesMustInstantiateCollections()
-        {
-            var source = WithOctopusTypes(@"
-namespace Octopus.Core.Features.ServerTasks.MessageContracts
-{
-    /// <summary>request</summary>
-    public class SimpleRequest: IRequest<SimpleRequest, SimpleResponse> 
-    {
-        protected SimpleRequest() { }
-        public SimpleRequest(List<string> requiredStringList)
-        {
-            RequiredStringListProperty = requiredStringList;
-        }
+        [Optional]
+        public int {|#1:IntProperty|} { get; set; } // NOT OK, must be nullable or initialized.
 
-        [Required]
-        public List<string> RequiredStringListProperty { get; set; } // should not fire on this; collection required so another check will ensure it's assigned in the constructor
+        [Optional]
+        public int? NullableIntProperty { get; set; } // OK
+
+        [Optional]
+        public int InitializedIntProperty { get; set; } = 10; // OK
+
+        [Optional]
+        public int InitializedIntPropertyEmpty { get; set; } = 0; // OK
 
         [Optional]
-        public List<string> {|#0:StringListProperty|} { get; set; } // should fire on this; collection is nonnullable
+        public bool {|#2:BoolProperty|} { get; set; } // NOT OK, must be nullable or initialized
 
         [Optional]
-        public string[] {|#1:StringArrayProperty|} { get; set; } // should fire on this; collection is nonnullable
+        public bool? NullableBoolProperty { get; set; } // OK
 
         [Optional]
-        public List<string>? NullableStringListProperty { get; set; } // should not fire on this; collection is nullable
+        public bool InitializedBoolProperty { get; set; } = true; // OK
 
         [Optional]
-        public string[]? NullableStringArrayProperty { get; set; } // should not fire on this; collection is nullable
+        public bool InitializedBoolPropertyEmpty { get; set; } = false; // OK
```

Comment:
nit: is `Empty` the right word here?

Reply:
yeah, it's more "CompilerDefaultValueForThisType", but that's too long. I'll change to 'TypeDefault' instead